diff --git a/src/main/java/org/apache/maven/changelog/ChangeLogReport.java b/src/main/java/org/apache/maven/changelog/ChangeLogReport.java
index 5c000d5..4340e49 100644
--- a/src/main/java/org/apache/maven/changelog/ChangeLogReport.java
+++ b/src/main/java/org/apache/maven/changelog/ChangeLogReport.java
@@ -98,7 +98,7 @@ public class ChangeLogReport
     private List tags;
 
     /**
-     * Used to specify the date format of log entries to retrieve. This format will also be reflected in the reports.
+     * Used to specify the date format of the log entries that are retrieved from your SCM system.
      *
      * @parameter expression="${changelog.dateFormat}" default-value="yyyy-MM-dd"
      * @required