diff --git a/intg/src/main/java/org/apache/atlas/type/AtlasEntityType.java b/intg/src/main/java/org/apache/atlas/type/AtlasEntityType.java
index ce61548cb..2557bb31b 100644
--- a/intg/src/main/java/org/apache/atlas/type/AtlasEntityType.java
+++ b/intg/src/main/java/org/apache/atlas/type/AtlasEntityType.java
@@ -34,7 +34,6 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -335,7 +334,7 @@ public class AtlasEntityType extends AtlasStructType {
         attributes.put(relationshipType.getTypeName(), attribute);
     }
 
-    public Collection<String> getAttributeRelationshipTypes(String attributeName) {
+    public Set<String> getAttributeRelationshipTypes(String attributeName) {
         Map<String, AtlasAttribute> attributes = relationshipAttributes.get(attributeName);
 
         return attributes != null ? attributes.keySet() : null;
@@ -353,6 +352,10 @@ public class AtlasEntityType extends AtlasStructType {
         return typeQryStr;
     }
 
+    public boolean hasAttribute(String attributeName) {
+        return allAttributes.containsKey(attributeName);
+    }
+
     public boolean hasRelationshipAttribute(String attributeName) {
         return relationshipAttributes.containsKey(attributeName);
     }