diff --git a/intg/src/main/java/org/apache/atlas/model/typedef/AtlasEntityDef.java b/intg/src/main/java/org/apache/atlas/model/typedef/AtlasEntityDef.java
index 230fca18c..36bb3df1f 100644
--- a/intg/src/main/java/org/apache/atlas/model/typedef/AtlasEntityDef.java
+++ b/intg/src/main/java/org/apache/atlas/model/typedef/AtlasEntityDef.java
@@ -54,6 +54,10 @@ public class AtlasEntityDef extends AtlasStructDef implements java.io.Serializab
 
     private Set<String> superTypes;
 
+    // subTypes field below is derived from 'superTypes' specified in all AtlasEntityDef
+    // this value is ignored during create & update operations
+    private Set<String> subTypes;
+
 
     public AtlasEntityDef() {
         this(null, null, null, null, null, null);
@@ -109,6 +113,14 @@ public class AtlasEntityDef extends AtlasStructDef implements java.io.Serializab
         }
     }
 
+    public Set<String> getSubTypes() {
+        return subTypes;
+    }
+
+    public void setSubTypes(Set<String> subTypes) {
+        this.subTypes = subTypes;
+    }
+
     public boolean hasSuperType(String typeName) {
         return hasSuperType(superTypes, typeName);
     }