diff --git a/maven-project-info-reports-plugin/src/main/java/org/apache/maven/report/projectinfo/TeamListReport.java b/maven-project-info-reports-plugin/src/main/java/org/apache/maven/report/projectinfo/TeamListReport.java
index 6164986b6..7aa8778fd 100644
--- a/maven-project-info-reports-plugin/src/main/java/org/apache/maven/report/projectinfo/TeamListReport.java
+++ b/maven-project-info-reports-plugin/src/main/java/org/apache/maven/report/projectinfo/TeamListReport.java
@@ -127,8 +127,6 @@ public class TeamListReport
 
         private final boolean showAvatarImages;
 
-        private static final String[] EMPTY_STRING_ARRAY = new String[0];
-
         TeamListRenderer( Sink sink, Model model, I18N i18n, Locale locale, Log log, boolean showAvatarImages )
         {
             super( sink, i18n, locale );
@@ -437,15 +435,6 @@ public class TeamListReport
             }
         }
 
-        private String img( String src )
-        {
-            if ( src == null )
-            {
-                return "";
-            }
-            return "<img src='" + src + "'/>";
-        }
-
         /**
          * @param requiredHeaders
          * @return