diff --git a/ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java b/ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java
index f61612649c..f24908ab23 100644
--- a/ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java
+++ b/ambari-server/src/main/java/org/apache/ambari/server/configuration/Configuration.java
@@ -295,7 +295,6 @@ public class Configuration {
     configsMap.put(SRVR_CRT_PASS_KEY, randStr);
 
     loadSSLParams();
-    loadCredentialProvider();
   }
 
   /**