diff --git a/src/main/java/org/apache/maven/plugin/assembly/utils/AssemblyFileUtils.java b/src/main/java/org/apache/maven/plugin/assembly/utils/AssemblyFileUtils.java
index 267efa7d..b04b913e 100644
--- a/src/main/java/org/apache/maven/plugin/assembly/utils/AssemblyFileUtils.java
+++ b/src/main/java/org/apache/maven/plugin/assembly/utils/AssemblyFileUtils.java
@@ -192,7 +192,7 @@ public final class AssemblyFileUtils
             }
             else
             {
-                throw new AssemblyFormattingException( "Illlegal lineEnding specified: '" + lineEnding + "'" );
+                throw new AssemblyFormattingException( "Illegal lineEnding specified: '" + lineEnding + "'" );
             }
         }
 