diff --git a/camel-core/src/main/java/org/apache/camel/ExchangePattern.java b/camel-core/src/main/java/org/apache/camel/ExchangePattern.java
index 56d5be2bc15..60de5db9949 100644
--- a/camel-core/src/main/java/org/apache/camel/ExchangePattern.java
+++ b/camel-core/src/main/java/org/apache/camel/ExchangePattern.java
@@ -32,6 +32,8 @@ import javax.xml.bind.annotation.XmlType;
 public enum ExchangePattern {
     InOnly, RobustInOnly, InOut, InOptionalOut, OutOnly, RobustOutOnly, OutIn, OutOptionalIn;
 
+    // TODO: We should deprecate and only support InOnly, InOut, and InOptionalOut
+
     protected static final Map<String, ExchangePattern> MAP = new HashMap<String, ExchangePattern>();
 
     /**