diff --git a/maven-enforcer-plugin/src/main/java/org/apache/maven/plugin/enforcer/EnforceMojo.java b/maven-enforcer-plugin/src/main/java/org/apache/maven/plugin/enforcer/EnforceMojo.java
index c29fc5880..f2dc03b2f 100644
--- a/maven-enforcer-plugin/src/main/java/org/apache/maven/plugin/enforcer/EnforceMojo.java
+++ b/maven-enforcer-plugin/src/main/java/org/apache/maven/plugin/enforcer/EnforceMojo.java
@@ -38,6 +38,7 @@ import org.apache.maven.shared.enforcer.rule.api.EnforcerRuleHelper;
  * @goal enforce
  * @author <a href="mailto:brianf@apache.org">Brian Fox</a>
  * @phase validate
+ * @requiresDependencyResolution test
  * @version $Id$
  */
 public class EnforceMojo