diff --git a/ql/src/java/org/apache/hadoop/hive/ql/lockmgr/HiveLockObject.java b/ql/src/java/org/apache/hadoop/hive/ql/lockmgr/HiveLockObject.java
index 82dc89803b..fff03df473 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/lockmgr/HiveLockObject.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/lockmgr/HiveLockObject.java
@@ -22,6 +22,7 @@
 
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.apache.hadoop.hive.common.StringInternUtils;
 import org.apache.hadoop.hive.metastore.MetaStoreUtils;
 import org.apache.hadoop.hive.ql.metadata.DummyPartition;
 import org.apache.hadoop.hive.ql.metadata.Hive;
@@ -53,9 +54,10 @@ public HiveLockObjectData(String queryId,
         String lockMode,
         String queryStr) {
       this.queryId = removeDelimiter(queryId);
-      this.lockTime = removeDelimiter(lockTime);
+      this.lockTime = StringInternUtils.internIfNotNull(removeDelimiter(lockTime));
       this.lockMode = removeDelimiter(lockMode);
-      this.queryStr = removeDelimiter(queryStr == null ? null : queryStr.trim());
+      this.queryStr = StringInternUtils.internIfNotNull(
+          removeDelimiter(queryStr == null ? null : queryStr.trim()));
     }
 
     /**
@@ -71,9 +73,9 @@ public HiveLockObjectData(String data) {
 
       String[] elem = data.split(":");
       queryId = elem[0];
-      lockTime = elem[1];
+      lockTime = StringInternUtils.internIfNotNull(elem[1]);
       lockMode = elem[2];
-      queryStr = elem[3];
+      queryStr = StringInternUtils.internIfNotNull(elem[3]);
       if (elem.length >= 5) {
         clientIp = elem[4];
       }
@@ -178,12 +180,12 @@ public HiveLockObject() {
 
   public HiveLockObject(String path, HiveLockObjectData lockData) {
     this.pathNames = new String[1];
-    this.pathNames[0] = path;
+    this.pathNames[0] = StringInternUtils.internIfNotNull(path);
     this.data = lockData;
   }
 
   public HiveLockObject(String[] paths, HiveLockObjectData lockData) {
-    this.pathNames = paths;
+    this.pathNames = StringInternUtils.internStringsInArray(paths);
     this.data = lockData;
   }
 