diff --git a/dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/math/BeamSqlMathBinaryExpression.java b/dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/math/BeamSqlMathBinaryExpression.java
index 2d444f8750..05250c03c4 100644
--- a/dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/math/BeamSqlMathBinaryExpression.java
+++ b/dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/math/BeamSqlMathBinaryExpression.java
@@ -31,8 +31,8 @@ import org.apache.calcite.sql.type.SqlTypeName;
  */
 public abstract class BeamSqlMathBinaryExpression extends BeamSqlExpression {
 
-  public BeamSqlMathBinaryExpression(List<BeamSqlExpression> operands) {
-    super(operands, SqlTypeName.ANY);
+  public BeamSqlMathBinaryExpression(List<BeamSqlExpression> operands, SqlTypeName outputType) {
+    super(operands, outputType);
   }
 
   @Override public boolean accept() {