diff --git a/library/src/main/java/com/datatorrent/lib/algo/AllAfterMatchStringValueMap.java b/library/src/main/java/com/datatorrent/lib/algo/AllAfterMatchStringValueMap.java
index 4d8ba8123..02b9c06c1 100644
--- a/library/src/main/java/com/datatorrent/lib/algo/AllAfterMatchStringValueMap.java
+++ b/library/src/main/java/com/datatorrent/lib/algo/AllAfterMatchStringValueMap.java
@@ -25,7 +25,7 @@ import java.util.Map;
 
 /**
  *
- * A compare operation is done on input tuple based on the property "key",
+ * A compare metric is done on input tuple based on the property "key",
  * "value", and "cmp". All tuples are emitted (inclusive) once a match is made.
  * The comparison is done by getting double value from the Number.
  * <p>