diff --git a/src/java/org/apache/commons/math/linear/QRDecompositionImpl.java b/src/java/org/apache/commons/math/linear/QRDecompositionImpl.java
index 1e01798ac..cf2a66382 100644
--- a/src/java/org/apache/commons/math/linear/QRDecompositionImpl.java
+++ b/src/java/org/apache/commons/math/linear/QRDecompositionImpl.java
@@ -205,7 +205,7 @@ public class QRDecompositionImpl implements QRDecomposition {
     public RealMatrix getQT()
         throws IllegalStateException {
 
-            if (cachedQT == null) {
+        if (cachedQT == null) {
 
             checkDecomposed();
 