diff --git a/server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java b/server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java
index 4617299fa..d04303b5f 100644
--- a/server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java
+++ b/server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java
@@ -421,7 +421,7 @@ public class SimpleGarbageCollector implements Iface {
                   TableState tableState = TableManager.getInstance().getTableState(tableId);
                   if (tableState != null && tableState != TableState.DELETING) {
                     // clone directories don't always exist
-                    if (!tabletDir.startsWith("c-"))
+                    if (!tabletDir.startsWith(Constants.CLONE_PREFIX))
                       log.debug("File doesn't exist: " + fullPath);
                   }
                 } else {