diff --git a/sopremo/sopremo-common/src/main/java/eu/stratosphere/sopremo/pact/CsvInputFormat.java b/sopremo/sopremo-common/src/main/java/eu/stratosphere/sopremo/pact/CsvInputFormat.java
index de9a353213..76cfea965f 100644
--- a/sopremo/sopremo-common/src/main/java/eu/stratosphere/sopremo/pact/CsvInputFormat.java
+++ b/sopremo/sopremo-common/src/main/java/eu/stratosphere/sopremo/pact/CsvInputFormat.java
@@ -2,13 +2,7 @@ package eu.stratosphere.sopremo.pact;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
-import java.io.InputStreamReader;
 import java.nio.charset.Charset;
-import java.util.Arrays;
-
-import org.codehaus.jackson.JsonNode;
-import org.codehaus.jackson.node.JsonNodeFactory;
-import org.codehaus.jackson.node.ObjectNode;
 
 import com.csvreader.CsvReader;
 
@@ -16,6 +10,7 @@ import eu.stratosphere.nephele.configuration.Configuration;
 import eu.stratosphere.nephele.fs.FileInputSplit;
 import eu.stratosphere.pact.common.io.TextInputFormat;
 import eu.stratosphere.pact.common.type.KeyValuePair;
+import eu.stratosphere.sopremo.jsondatamodel.ObjectNode;
 import eu.stratosphere.sopremo.pact.PactJsonObject.Key;
 
 public class CsvInputFormat extends TextInputFormat<PactJsonObject.Key, PactJsonObject> {