diff --git a/sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/logging/DataflowWorkerLoggingHandler.java b/sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/logging/DataflowWorkerLoggingHandler.java
index 11b79f774e..6387048aff 100644
--- a/sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/logging/DataflowWorkerLoggingHandler.java
+++ b/sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/worker/logging/DataflowWorkerLoggingHandler.java
@@ -120,11 +120,10 @@ public class DataflowWorkerLoggingHandler extends Handler {
 
   /**
    * Check if a LogRecord will be logged.
-   * <p>
-   * This method checks if the <tt>LogRecord</tt> has an appropriate level and
+   *
+   * <p>This method checks if the <tt>LogRecord</tt> has an appropriate level and
    * whether it satisfies any <tt>Filter</tt>.  It will also return false if
    * the handler has been closed, or the LogRecord is null.
-   * <p>
    */
   @Override
   public boolean isLoggable(LogRecord record) {