diff --git a/src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java b/src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java
index 9b8e536519..c9433a430c 100644
--- a/src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java
+++ b/src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java
@@ -627,7 +627,8 @@ public class DefaultGroovyMethods {
     // Number based methods
     //-------------------------------------------------------------------------
     public static String plus(String left, Object value) {
-        return left + toString(value);
+        return left + value;
+        //return left + toString(value);
     }
 
     public static String minus(String left, Object value) {