diff --git a/src/java/org/apache/ddlutils/io/DatabaseIO.java b/src/java/org/apache/ddlutils/io/DatabaseIO.java
index 8a09afc..9662a9d 100644
--- a/src/java/org/apache/ddlutils/io/DatabaseIO.java
+++ b/src/java/org/apache/ddlutils/io/DatabaseIO.java
@@ -1046,7 +1046,7 @@ public class DatabaseIO
      */
     private void writeAttribute(PrettyPrintingXmlWriter xmlWriter, QName qName, String value) throws DdlUtilsXMLException
     {
-        if (!StringUtils.isEmpty(value))
+        if (value != null)
         {
             xmlWriter.writeAttribute(null, qName.getLocalPart(), value);
         }