diff --git a/storage/sis-storage/src/main/java/org/apache/sis/internal/storage/MetadataBuilder.java b/storage/sis-storage/src/main/java/org/apache/sis/internal/storage/MetadataBuilder.java
index ae639fe92..d80516dad 100644
--- a/storage/sis-storage/src/main/java/org/apache/sis/internal/storage/MetadataBuilder.java
+++ b/storage/sis-storage/src/main/java/org/apache/sis/internal/storage/MetadataBuilder.java
@@ -1946,10 +1946,7 @@ parse:      for (int i = 0; i < length;) {
                     if (axisType.isPresent()) {
                         setAxisName(i, axisType.get());
                     }
-                    final long size = extent.getSize(i);
-                    if (size >= 0 && size <= Integer.MAX_VALUE) {
-                        setAxisLength(i, (int) size);
-                    }
+                    setAxisSize(i, extent.getSize(i));
                 }
             }
             if (addResolution && grid.isDefined(GridGeometry.RESOLUTION)) {
@@ -2179,8 +2176,10 @@ parse:      for (int i = 0; i < length;) {
      * @param  dimension  the axis dimension.
      * @param  length     number of cell values along the given dimension.
      */
-    public final void setAxisLength(final int dimension, final int length) {
-        axis(dimension).setDimensionSize(shared(length));
+    public final void setAxisSize(final int dimension, final long length) {
+        if (length >= 0) {
+            axis(dimension).setDimensionSize(shared(length > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) length));
+        }
     }
 
     /**