diff --git a/sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java b/sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java
index 574ba0c493..9b273f8621 100644
--- a/sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java
+++ b/sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java
@@ -18,7 +18,6 @@
 package org.apache.beam.sdk.io;
 
 import javax.annotation.Nullable;
-import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.transforms.display.DisplayData;
 import org.apache.beam.sdk.util.NameUtils;
@@ -159,11 +158,6 @@ public class Read {
       return new BoundedReadFromUnboundedSource<T>(source, Long.MAX_VALUE, maxReadTime);
     }
 
-    @Override
-    protected Coder<T> getDefaultOutputCoder() {
-      return source.getOutputCoder();
-    }
-
     @Override
     public final PCollection<T> expand(PBegin input) {
       source.validate();