diff --git a/features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java b/features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java
index e71386dd39..aded8af4ac 100644
--- a/features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java
+++ b/features/core/src/main/java/org/apache/karaf/features/internal/service/Deployer.java
@@ -727,8 +727,8 @@ public class Deployer {
                             && isUpdateable(resource) && !deployment.bundleChecksums.containsKey(bundle.getBundleId())) {
                         deployment.bundleChecksums.put(bundle.getBundleId(), crc);
                     }
-                    int startLevel = startLevels.get(resource);
-                    if (startLevel != dstate.initialBundleStartLevel) {
+                    Integer startLevel = startLevels.get(resource);
+                    if (startLevel != null && startLevel != dstate.initialBundleStartLevel) {
                         callback.setBundleStartLevel(bundle, startLevel);
                     }
                     FeatureState reqState = states.get(resource);