diff --git a/geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/CommitCommand.java b/geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/CommitCommand.java
index 3ded490b0f..d475389392 100644
--- a/geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/CommitCommand.java
+++ b/geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/CommitCommand.java
@@ -18,6 +18,7 @@ import java.io.IOException;
 import java.util.concurrent.TimeoutException;
 
 import org.apache.geode.CancelException;
+import org.apache.geode.annotations.Immutable;
 import org.apache.geode.cache.TransactionInDoubtException;
 import org.apache.geode.distributed.DistributedMember;
 import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
@@ -40,6 +41,7 @@ import org.apache.geode.internal.security.SecurityService;
  */
 public class CommitCommand extends BaseCommand {
 
+  @Immutable
   private static final CommitCommand singleton = new CommitCommand();
 
   public static Command getCommand() {