diff --git a/components/camel-jcache/src/main/java/org/apache/camel/component/jcache/JCacheConfiguration.java b/components/camel-jcache/src/main/java/org/apache/camel/component/jcache/JCacheConfiguration.java
index 05dd5a68c3c..43b7e4d31b4 100644
--- a/components/camel-jcache/src/main/java/org/apache/camel/component/jcache/JCacheConfiguration.java
+++ b/components/camel-jcache/src/main/java/org/apache/camel/component/jcache/JCacheConfiguration.java
@@ -38,17 +38,16 @@ import org.apache.camel.util.EndpointHelper;
 
 @UriParams
 public class JCacheConfiguration {
-    @UriParam(label = "advanced")
-    @Metadata(required = "true")
+    @UriParam(label = "common")
     private String cachingProvider;
 
-    @UriParam(label = "advanced")
+    @UriParam
     private Configuration cacheConfiguration;
 
-    @UriParam(label = "advanced")
+    @UriParam
     private Properties cacheConfigurationProperties;
 
-    @UriParam(label = "advanced")
+    @UriParam
     private String configurationUri;
 
     @UriParam(label = "advanced")