diff --git a/core/sis-utility/src/main/java/org/apache/sis/internal/util/DoubleDouble.java b/core/sis-utility/src/main/java/org/apache/sis/internal/util/DoubleDouble.java
index 965f5e6f3..60477c4af 100644
--- a/core/sis-utility/src/main/java/org/apache/sis/internal/util/DoubleDouble.java
+++ b/core/sis-utility/src/main/java/org/apache/sis/internal/util/DoubleDouble.java
@@ -21,6 +21,9 @@ import org.apache.sis.math.MathFunctions;
 import org.apache.sis.math.DecimalFunctions;
 // No BigDecimal dependency - see class javadoc
 
+// Branch-dependent imports
+import org.apache.sis.internal.jdk8.JDK8;
+
 
 /**
  * Basic arithmetic methods for extended precision numbers using the <cite>double-double</cite> algorithm.
@@ -284,7 +287,7 @@ public final class DoubleDouble extends Number {
     @Override public double doubleValue() {return value;}
     @Override public float  floatValue()  {return (float) value;}
     @Override public long   longValue()   {return Math.round(value);}
-    @Override public int    intValue()    {return (int) longValue();}
+    @Override public int    intValue()    {return JDK8.toIntExact(longValue());}
 
     /**
      * Suggests an {@link #error} for the given value. The {@code DoubleDouble} class contains a hard-coded list