diff --git a/osworkflow/src/java/org/apache/fulcrum/osworkflow/WorkflowService.java b/osworkflow/src/java/org/apache/fulcrum/osworkflow/WorkflowService.java
index 800c1f94..e8730677 100644
--- a/osworkflow/src/java/org/apache/fulcrum/osworkflow/WorkflowService.java
+++ b/osworkflow/src/java/org/apache/fulcrum/osworkflow/WorkflowService.java
@@ -71,14 +71,21 @@ public interface WorkflowService
     /** Avalon role - used to id the component within the manager */
     String ROLE = WorkflowService.class.getName();
 
-    /** Retrives a workflow based on the caller */
+    /** Retrives a workflow based on the caller 
+     * @param caller the value of the caller
+     * @return the Workflow for this caller
+     */
     public Workflow retrieveWorkflow(String caller);
     
-    /** For a specific caller and status, return all the workflows. */
+    /** For a specific caller and status, return all the workflows.
+     * 
+     * @param caller the value of the caller
+     * @param status the status, defined in the workflow xml file
+     * @return an array of longs of the workflow id's
+     * @throws com.opensymphony.workflow.WorkflowException if there is a problem
+     */
     public long[] retrieveWorkflows(String caller, String status) throws WorkflowException;
     
-    /** Retrieves a workflow instance */
-   // public WorkflowInstance retrieveWorkflowInstance(Workflow workflow, long id);
     
 
 