diff --git a/library/src/main/java/com/datatorrent/lib/logs/FilteredLineToTokenArrayList.java b/library/src/main/java/com/datatorrent/lib/logs/FilteredLineToTokenArrayList.java
index a43e46131..1934e17a0 100644
--- a/library/src/main/java/com/datatorrent/lib/logs/FilteredLineToTokenArrayList.java
+++ b/library/src/main/java/com/datatorrent/lib/logs/FilteredLineToTokenArrayList.java
@@ -44,8 +44,8 @@ import java.util.HashMap;
  * <br>
  * </p>
  * @displayName Filtered Line To Token ArrayList
- * @category Logs
- * @tags string
+ * @category Tuple Converters
+ * @tags filter, arraylist, string
  *
  * @since 0.3.2
  */