diff --git a/core/src/main/java/org/apache/mahout/cf/taste/impl/model/GenericUserPreferenceArray.java b/core/src/main/java/org/apache/mahout/cf/taste/impl/model/GenericUserPreferenceArray.java
index 7f7304673..0f1a18149 100644
--- a/core/src/main/java/org/apache/mahout/cf/taste/impl/model/GenericUserPreferenceArray.java
+++ b/core/src/main/java/org/apache/mahout/cf/taste/impl/model/GenericUserPreferenceArray.java
@@ -114,6 +114,14 @@ public final class GenericUserPreferenceArray implements PreferenceArray {
   public void setItemID(int i, long itemID) {
     ids[i] = itemID;
   }
+
+  /**
+   * @return all item IDs
+   */
+  @Override
+  public long[] getIDs() {
+    return ids;
+  }
   
   @Override
   public float getValue(int i) {