diff --git a/components/camel-gae/src/main/java/org/apache/camel/component/gae/mail/GMailEndpoint.java b/components/camel-gae/src/main/java/org/apache/camel/component/gae/mail/GMailEndpoint.java
index 93e25a8810b..1562601ad87 100644
--- a/components/camel-gae/src/main/java/org/apache/camel/component/gae/mail/GMailEndpoint.java
+++ b/components/camel-gae/src/main/java/org/apache/camel/component/gae/mail/GMailEndpoint.java
@@ -40,7 +40,7 @@ public class GMailEndpoint extends DefaultEndpoint implements OutboundBindingSup
 
     private OutboundBinding<GMailEndpoint, Message, Void> outboundBinding;
     private MailService mailService;
-    @UriPath @Metadata(required = "true")
+    @UriPath(description = "The email of the sender") @Metadata(required = "true")
     private String sender;
     @UriParam
     private String subject;
@@ -77,6 +77,9 @@ public class GMailEndpoint extends DefaultEndpoint implements OutboundBindingSup
         return subject;
     }
 
+    /**
+     * Subject of the email.
+     */
     public void setSubject(String subject) {
         this.subject = subject;
     }
@@ -85,6 +88,9 @@ public class GMailEndpoint extends DefaultEndpoint implements OutboundBindingSup
         return to;
     }
 
+    /**
+     * To-receiver of the email. This can be a single receiver or a comma-separated list of receivers.
+     */
     public void setTo(String to) {
         this.to = to;
     }
@@ -93,6 +99,9 @@ public class GMailEndpoint extends DefaultEndpoint implements OutboundBindingSup
         return cc;
     }
 
+    /**
+     * Cc-receiver of the email. This can be a single receiver or a comma-separated list of receivers.
+     */
     public void setCc(String cc) {
         this.cc = cc;
     }
@@ -101,6 +110,9 @@ public class GMailEndpoint extends DefaultEndpoint implements OutboundBindingSup
         return bcc;
     }
 
+    /**
+     * Bcc-receiver of the email. This can be a single receiver or a comma-separated list of receivers.
+     */
     public void setBcc(String bcc) {
         this.bcc = bcc;
     }