diff --git a/maven-release-plugin/src/main/java/org/apache/maven/plugins/release/StageReleaseMojo.java b/maven-release-plugin/src/main/java/org/apache/maven/plugins/release/StageReleaseMojo.java
index 7fddf46..92bfb31 100644
--- a/maven-release-plugin/src/main/java/org/apache/maven/plugins/release/StageReleaseMojo.java
+++ b/maven-release-plugin/src/main/java/org/apache/maven/plugins/release/StageReleaseMojo.java
@@ -24,6 +24,7 @@ import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.shared.release.ReleaseExecutionException;
 import org.apache.maven.shared.release.ReleaseFailureException;
 import org.apache.maven.shared.release.config.ReleaseDescriptor;
+import org.apache.maven.shared.release.util.ReleaseUtil;
 import org.codehaus.plexus.util.StringUtils;
 
 import java.io.File;
@@ -122,7 +123,8 @@ public class StageReleaseMojo
 
             releaseDescriptor.setCheckoutDirectory( workingDirectory.getAbsolutePath() );
             releaseDescriptor.setUseReleaseProfile( useReleaseProfile );
-
+            releaseDescriptor.setRootProjectPath( ReleaseUtil.getRootProjectPath( project ) );
+            
             if ( goals == null )
             {
                 // set default