diff --git a/jena-tdb/src/main/java/com/hp/hpl/jena/tdb/base/file/LocationLock.java b/jena-tdb/src/main/java/com/hp/hpl/jena/tdb/base/file/LocationLock.java
index d0e8d9ce50..c5aeaf2f64 100644
--- a/jena-tdb/src/main/java/com/hp/hpl/jena/tdb/base/file/LocationLock.java
+++ b/jena-tdb/src/main/java/com/hp/hpl/jena/tdb/base/file/LocationLock.java
@@ -174,11 +174,11 @@ public class LocationLock {
             // Check if the owner is alive
             if (ProcessUtils.isAlive(owner))
                 throw new TDBException(
-                        "The location "
+                        "Location "
                                 + location.getDirectoryPath()
                                 + " is currently locked by PID "
                                 + owner
-                                + ".  TDB databases do not permit concurrent usage across JVMs so in order to prevent corruption you cannot open this location from the JVM that does not own the lock for the dataset");
+                                + ".  TDB databases do not permit concurrent usage across JVMs so in order to prevent possible data corruption you cannot open this location from the JVM that does not own the lock for the dataset");
 
             // Otherwise the previous owner is dead so we can take the lock
             takeLock(ProcessUtils.getPid(NO_OWNER));