diff --git a/geode-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java b/geode-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java
index 84ba926ee3..76a7bad405 100755
--- a/geode-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java
+++ b/geode-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java
@@ -1983,6 +1983,7 @@ public class GemFireCacheImpl implements InternalCache, ClientCache, HasCachePer
         }
       }
 
+      this.keepAlive = keepalive;
       isClosing = true;
       logger.info(LocalizedMessage.create(LocalizedStrings.GemFireCache_0_NOW_CLOSING, this));
 
@@ -1999,7 +2000,6 @@ public class GemFireCacheImpl implements InternalCache, ClientCache, HasCachePer
       }
       TXStateProxy tx = null;
       try {
-        this.keepAlive = keepalive;
 
         if (this.txMgr != null) {
           tx = this.txMgr.internalSuspend();