diff --git a/java/org/apache/coyote/http2/StreamProcessor.java b/java/org/apache/coyote/http2/StreamProcessor.java
index c0555d6e39..ecbeaa9369 100644
--- a/java/org/apache/coyote/http2/StreamProcessor.java
+++ b/java/org/apache/coyote/http2/StreamProcessor.java
@@ -250,8 +250,8 @@ class StreamProcessor extends AbstractProcessor {
 
 
     @Override
-    protected final boolean isReady() {
-        return stream.isReady();
+    protected final boolean isReadyForWrite() {
+        return stream.isReadyForWrite();
     }
 
 
@@ -373,7 +373,7 @@ class StreamProcessor extends AbstractProcessor {
             // has been emptied then the code below will call
             // dispatch() which will enable the
             // Response to respond to this event.
-            if (stream.isReady()) {
+            if (stream.isReadyForWrite()) {
                 // Unexpected
                 throw new IllegalStateException();
             }