diff --git a/tools/common/src/main/java/org/apache/cxf/tools/common/model/JAnnotation.java b/tools/common/src/main/java/org/apache/cxf/tools/common/model/JAnnotation.java
index dd48ebe358..4ae0836853 100644
--- a/tools/common/src/main/java/org/apache/cxf/tools/common/model/JAnnotation.java
+++ b/tools/common/src/main/java/org/apache/cxf/tools/common/model/JAnnotation.java
@@ -86,7 +86,7 @@ public final class JAnnotation {
     }
 
     public String toString() {
-        StringBuffer sb = new StringBuffer();
+        StringBuilder sb = new StringBuilder();
         sb.append("@");
         sb.append(type.getSimpleName());
         if (getElements().isEmpty()) {