diff --git a/ldap/model/src/main/java/org/apache/directory/api/ldap/model/ldif/LdifRevertor.java b/ldap/model/src/main/java/org/apache/directory/api/ldap/model/ldif/LdifRevertor.java
index 745a6e601..e99962315 100644
--- a/ldap/model/src/main/java/org/apache/directory/api/ldap/model/ldif/LdifRevertor.java
+++ b/ldap/model/src/main/java/org/apache/directory/api/ldap/model/ldif/LdifRevertor.java
@@ -470,7 +470,7 @@ public final class LdifRevertor
         newDn = newDn.add( newRdn );
 
         List<LdifEntry> entries = new ArrayList<>( 1 );
-        LdifEntry reverted = new LdifEntry();
+        LdifEntry reverted;
 
         // Start with the cases here
         if ( newRdn.size() == 1 )