diff --git a/wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/ScpWagon.java b/wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/ScpWagon.java
index d23d9227..a30a567d 100644
--- a/wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/ScpWagon.java
+++ b/wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/ScpWagon.java
@@ -233,13 +233,16 @@ public void executeCommand( String command )
     public void put( File source, String resourceName )
         throws TransferFailedException, ResourceDoesNotExistException, AuthorizationException
     {
-
         String basedir = getRepository().getBasedir();
 
         resourceName = StringUtils.replace( resourceName, "\\", "/" );
         String dir = PathUtils.dirname( resourceName );
         dir = StringUtils.replace( dir, "\\", "/" );
 
+        Resource resource = new Resource( resourceName );
+
+        firePutInitiated( resource, source );
+
         String mkdirCmd = "mkdir -p " + basedir + "/" + dir + "\n";
 
         executeCommand( mkdirCmd );
@@ -299,8 +302,6 @@ public void put( File source, String resourceName )
                 throw new TransferFailedException( "ACK check failed" );
             }
 
-            Resource resource = new Resource( resourceName );
-
             putTransfer( resource, source, out, false );
 
             byte[] buf = new byte[1024];