diff --git a/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/CseClientHttpRequestFactory.java b/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/CseClientHttpRequestFactory.java
index e89bd3dab..23d3ce0b7 100644
--- a/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/CseClientHttpRequestFactory.java
+++ b/provider-springmvc/src/main/java/io/servicecomb/provider/springmvc/reference/CseClientHttpRequestFactory.java
@@ -26,7 +26,7 @@ import org.springframework.http.client.ClientHttpRequestFactory;
 /**
  * <一句话功能简述>
  * <功能详细描述>
- * @author   
+ *
  * @version  [版本号, 2017年1月14日]
  * @see  [相关类/方法]
  * @since  [产品/模块版本]