diff --git a/blueprints/blueprints-core/src/main/java/com/tinkerpop/blueprints/Property.java b/blueprints/blueprints-core/src/main/java/com/tinkerpop/blueprints/Property.java
index 5181b8f24e..cd187b86d4 100644
--- a/blueprints/blueprints-core/src/main/java/com/tinkerpop/blueprints/Property.java
+++ b/blueprints/blueprints-core/src/main/java/com/tinkerpop/blueprints/Property.java
@@ -116,7 +116,7 @@ public abstract interface Property<V> {
         }
 
         public static UnsupportedOperationException dataTypeOfPropertyValueNotSupported(final Object val) {
-            throw new UnsupportedOperationException(String.format("Property value [%s] is of type %s which is not supported by this Graph implementation", val, val.getClass()));
+            throw new UnsupportedOperationException(String.format("Property value [%s] is of type %s is not supported", val, val.getClass()));
         }
     }
 