diff --git a/library/src/main/java/com/datatorrent/lib/bucket/BucketManager.java b/library/src/main/java/com/datatorrent/lib/bucket/BucketManager.java
index ff5b2e655..73213a778 100644
--- a/library/src/main/java/com/datatorrent/lib/bucket/BucketManager.java
+++ b/library/src/main/java/com/datatorrent/lib/bucket/BucketManager.java
@@ -90,7 +90,7 @@ public interface BucketManager<T extends Bucketable>
   /**
    * <p>
    * Returns the bucket in memory corresponding to a bucket key.
-   * If the bucket is not created yet, it will return null.</br>
+   * If the bucket is not created yet, it will return null.
    * A bucket is created:
    * <ul>
    * <li> When the operator requests to load a bucket from store.</li>