diff --git a/repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java b/repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
index 3c7f63bbf..67b5362c9 100755
--- a/repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
+++ b/repository/src/main/java/org/apache/atlas/repository/graph/GraphBackedSearchIndexer.java
@@ -285,7 +285,7 @@ public class GraphBackedSearchIndexer implements SearchIndexer, ActiveStateChang
             } else if (isEnumType(atlasType)) {
                 createIndexes(management, propertyName, String.class, isUnique, cardinality, false, isIndexable);
             } else if (isStructType(atlasType)) {
-                AtlasStructDef structDef = typeRegistry.getStructDefByName(attributeDef.getName());
+                AtlasStructDef structDef = typeRegistry.getStructDefByName(attribTypeName);
                 updateIndexForTypeDef(management, structDef);
             }
         } catch (AtlasBaseException e) {