diff --git a/proton-j/proton/src/main/java/org/apache/qpid/proton/engine/impl/TransportImpl.java b/proton-j/proton/src/main/java/org/apache/qpid/proton/engine/impl/TransportImpl.java
index fa7f9c4c..87024134 100644
--- a/proton-j/proton/src/main/java/org/apache/qpid/proton/engine/impl/TransportImpl.java
+++ b/proton-j/proton/src/main/java/org/apache/qpid/proton/engine/impl/TransportImpl.java
@@ -394,7 +394,7 @@ public class TransportImpl extends EndpointImpl
             DeliveryImpl delivery = _connectionEndpoint.getTransportWorkHead();
             while(delivery != null && buffer.remaining() >= _maxFrameSize )
             {
-                if((delivery.getLink() instanceof SenderImpl) && delivery.isLocalStateChange() && delivery.getTransportDelivery() != null)
+                if((delivery.getLink() instanceof SenderImpl) && delivery.getTransportDelivery() != null)
                 {
                     TransportDelivery transportDelivery = delivery.getTransportDelivery();
                     Disposition disposition = new Disposition();
@@ -509,7 +509,7 @@ public class TransportImpl extends EndpointImpl
             DeliveryImpl delivery = _connectionEndpoint.getTransportWorkHead();
             while(delivery != null && buffer.remaining() >= _maxFrameSize)
             {
-                if((delivery.getLink() instanceof ReceiverImpl) && delivery.isLocalStateChange())
+                if((delivery.getLink() instanceof ReceiverImpl))
                 {
                     TransportDelivery transportDelivery = delivery.getTransportDelivery();
                     Disposition disposition = new Disposition();