diff --git a/foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/net/NetUtils.java b/foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/net/NetUtils.java
index b19d81d4c..5763ec84d 100644
--- a/foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/net/NetUtils.java
+++ b/foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/net/NetUtils.java
@@ -16,10 +16,12 @@
 
 package io.servicecomb.foundation.common.net;
 
+import java.io.IOException;
 import java.net.Inet4Address;
 import java.net.Inet6Address;
 import java.net.InetAddress;
 import java.net.NetworkInterface;
+import java.net.ServerSocket;
 import java.net.SocketException;
 import java.net.URI;
 import java.net.URISyntaxException;
@@ -184,4 +186,12 @@ public final class NetUtils {
         }
         return address;
     }
+
+    public static boolean canTcpListen(InetAddress address, int port) {
+        try (ServerSocket ss = new ServerSocket(port, 0, address)) {
+            return true;
+        } catch (IOException e) {
+            return false;
+        }
+    }
 }