diff --git a/src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java b/src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java
index e1ce6380e..4ef202c17 100644
--- a/src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java
+++ b/src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java
@@ -207,10 +207,6 @@ public class ConfigurationManager {
   static final String EXECUTOR_REQUIRED_WITH_DOCKER =
       "This scheduler is configured to require an executor for Docker-based tasks.";
 
-  @VisibleForTesting
-  static final String CONTAINER_AND_IMAGE_ARE_MUTUALLY_EXCLUSIVE =
-      "A task may not have both a Docker container and an image.";
-
   /**
    * Check validity of and populates defaults in a task configuration.  This will return a deep copy
    * of the provided task configuration with default configuration values applied, and configuration
@@ -311,10 +307,6 @@ public class ConfigurationManager {
               + containerType.get().toString());
     }
 
-    if (containerType.get() != Container._Fields.MESOS && config.isSetImage()) {
-      throw new TaskDescriptionException(CONTAINER_AND_IMAGE_ARE_MUTUALLY_EXCLUSIVE);
-    }
-
     ThriftBackfill.backfillTask(builder);
 
     String types = config.getResources().stream()