diff --git a/doxia-core/src/main/java/org/apache/maven/doxia/module/docbook/DocBookParser.java b/doxia-core/src/main/java/org/apache/maven/doxia/module/docbook/DocBookParser.java
index bdf2caa4..e0f32f25 100644
--- a/doxia-core/src/main/java/org/apache/maven/doxia/module/docbook/DocBookParser.java
+++ b/doxia-core/src/main/java/org/apache/maven/doxia/module/docbook/DocBookParser.java
@@ -35,7 +35,7 @@ import java.util.Stack;
  * Sink.
  *
  * @author <a href="mailto:jason@maven.org">Jason van Zyl</a>
- * @version $Id$
+ * @version $Id:DocBookParser.java 348605 2005-11-24 12:02:44 +1100 (Thu, 24 Nov 2005) brett $
  * @plexus.component role="org.apache.maven.doxia.parser.Parser"
  * role-hint="doc-book"
  */