diff --git a/tiles3/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReaderException.java b/tiles3/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReaderException.java
index 0b0ed846..644f38a5 100644
--- a/tiles3/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReaderException.java
+++ b/tiles3/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReaderException.java
@@ -55,7 +55,7 @@ public class DigesterDefinitionsReaderException extends TilesException {
      * @param e The exception to be wrapped.
      * @since 2.1.0
      */
-    public DigesterDefinitionsReaderException(Exception e) {
+    public DigesterDefinitionsReaderException(Throwable e) {
         super(e);
     }
 
@@ -66,7 +66,7 @@ public class DigesterDefinitionsReaderException extends TilesException {
      * @param e The exception to be wrapped.
      * @since 2.1.0
      */
-    public DigesterDefinitionsReaderException(String message, Exception e) {
+    public DigesterDefinitionsReaderException(String message, Throwable e) {
         super(message, e);
     }
 