diff --git a/core/src/main/java/org/eigenbase/util/ChunkList.java b/core/src/main/java/org/eigenbase/util/ChunkList.java
index 5e6c1ccb9..c1598495c 100644
--- a/core/src/main/java/org/eigenbase/util/ChunkList.java
+++ b/core/src/main/java/org/eigenbase/util/ChunkList.java
@@ -41,16 +41,22 @@
     }
   }
 
-  /** Creates an empty ChunkList. */
+  /**
+   * Creates an empty ChunkList.
+   */
   public ChunkList() {
   }
 
-  /** Creates a ChunkList whose contents are a given Collection. */
+  /**
+   * Creates a ChunkList whose contents are a given Collection.
+   */
   public ChunkList(Collection<E> collection) {
     addAll(collection);
   }
 
-  /** For debugging and testing. */
+  /**
+   * For debugging and testing.
+   */
   boolean isValid(boolean fail) {
     if ((first == null) != (last == null)) {
       assert !fail;