diff --git a/saga-core/src/main/java/io/servicecomb/saga/core/ForwardRecovery.java b/saga-core/src/main/java/io/servicecomb/saga/core/ForwardRecovery.java
index dc409c5..a61e905 100644
--- a/saga-core/src/main/java/io/servicecomb/saga/core/ForwardRecovery.java
+++ b/saga-core/src/main/java/io/servicecomb/saga/core/ForwardRecovery.java
@@ -1,11 +1,12 @@
 /*
- * Copyright 2017 Huawei Technologies Co., Ltd
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *      http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,