diff --git a/core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java b/core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java
index c20806ed0..19f4bcd32 100644
--- a/core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java
+++ b/core/src/main/java/org/apache/accumulo/core/cli/ClientOpts.java
@@ -104,14 +104,14 @@ public class ClientOpts extends Help {
   public String principal = System.getProperty("user.name");
   
   @Parameter(names = "-p", converter = PasswordConverter.class, description = "Connection password")
-  public Password password = new Password("secret");
+  public Password password = null;
   
   @Parameter(names = "--password", converter = PasswordConverter.class, description = "Enter the connection password", password = true)
   public Password securePassword = null;
   
   public AuthenticationToken getToken() {
     if (securePassword == null) {
-      if (password.value == null)
+      if (password == null)
         return null;
       return new PasswordToken(password.value);
     }