diff --git a/core/src/main/java/org/apache/accumulo/core/client/mapred/RangeInputSplit.java b/core/src/main/java/org/apache/accumulo/core/client/mapred/RangeInputSplit.java
index 3fd2ab0e9..29274aace 100644
--- a/core/src/main/java/org/apache/accumulo/core/client/mapred/RangeInputSplit.java
+++ b/core/src/main/java/org/apache/accumulo/core/client/mapred/RangeInputSplit.java
@@ -33,7 +33,7 @@ public class RangeInputSplit extends org.apache.accumulo.core.client.mapreduce.R
   public RangeInputSplit(RangeInputSplit split) throws IOException {
     super(split);
   }
-  
+
   protected RangeInputSplit(String table, String tableId, Range range, String[] locations) {
     super(table, tableId, range, locations);
   }