diff --git a/library/src/main/java/com/datatorrent/lib/codec/JavaSerializationStreamCodec.java b/library/src/main/java/com/datatorrent/lib/codec/JavaSerializationStreamCodec.java
index 8c7ff2e90..d029ecfea 100644
--- a/library/src/main/java/com/datatorrent/lib/codec/JavaSerializationStreamCodec.java
+++ b/library/src/main/java/com/datatorrent/lib/codec/JavaSerializationStreamCodec.java
@@ -1,11 +1,11 @@
-/*
- * Copyright (c) 2013 DataTorrent, Inc. ALL Rights Reserved.
+/**
+ * Copyright (C) 2015 DataTorrent, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *   http://www.apache.org/licenses/LICENSE-2.0
+ *         http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
@@ -23,7 +23,7 @@ import java.io.ObjectOutputStream;
 import java.io.Serializable;
 
 import com.datatorrent.api.StreamCodec;
-import com.datatorrent.common.util.Slice;
+import com.datatorrent.netlet.util.Slice;
 
 /**
  * This codec is used for serializing the objects of class which implements