diff --git a/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/container/ProcessFunctionContainer.java b/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/container/ProcessFunctionContainer.java
index 3653e1761..8d211f4ce 100644
--- a/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/container/ProcessFunctionContainer.java
+++ b/pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/container/ProcessFunctionContainer.java
@@ -28,6 +28,7 @@ import io.grpc.ManagedChannel;
 import io.grpc.ManagedChannelBuilder;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.functions.instance.InstanceConfig;
 import org.apache.pulsar.functions.proto.Function;
 import org.apache.pulsar.functions.proto.InstanceCommunication;
 import org.apache.pulsar.functions.proto.InstanceCommunication.FunctionStatus;