diff --git a/scm-api.old/src/main/java/org/apache/maven/scm/manager/ScmManager.java b/scm-api.old/src/main/java/org/apache/maven/scm/manager/ScmManager.java
index 867afba9..cd6bab1e 100644
--- a/scm-api.old/src/main/java/org/apache/maven/scm/manager/ScmManager.java
+++ b/scm-api.old/src/main/java/org/apache/maven/scm/manager/ScmManager.java
@@ -18,10 +18,7 @@
  */
 
 import org.apache.maven.scm.ScmException;
-import org.apache.maven.scm.Scm;
 import org.apache.maven.scm.command.Command;
-import org.apache.maven.scm.command.CommandWrapper;
-import org.apache.maven.scm.repository.Repository;
 import org.apache.maven.scm.repository.RepositoryInfo;
 
 public interface ScmManager