diff --git a/src/main/java/org/apache/directory/fortress/core/rbac/AdminMgrImpl.java b/src/main/java/org/apache/directory/fortress/core/rbac/AdminMgrImpl.java
index 8c3bce5f..e252dd81 100755
--- a/src/main/java/org/apache/directory/fortress/core/rbac/AdminMgrImpl.java
+++ b/src/main/java/org/apache/directory/fortress/core/rbac/AdminMgrImpl.java
@@ -71,7 +71,7 @@ import org.apache.directory.fortress.core.util.time.CUtil;
  * <p/>
  * This class is NOT thread safe if parent instance variables ({@link #contextId} or {@link #adminSess}) are set.
  *
- * @author Shawn McKinney
+ * @author <a href="mailto:dev@directory.apache.org">Apache Directory Project</a>
  */
 public final class AdminMgrImpl extends Manageable implements AdminMgr
 {