diff --git a/lib/commons-core/src/main/java/org/apache/olingo/commons/core/edm/primitivetype/EdmSingle.java b/lib/commons-core/src/main/java/org/apache/olingo/commons/core/edm/primitivetype/EdmSingle.java
index b6292775b..1fd98f07b 100644
--- a/lib/commons-core/src/main/java/org/apache/olingo/commons/core/edm/primitivetype/EdmSingle.java
+++ b/lib/commons-core/src/main/java/org/apache/olingo/commons/core/edm/primitivetype/EdmSingle.java
@@ -40,8 +40,7 @@ public final class EdmSingle extends SingletonPrimitiveType {
 
   @Override
   public boolean isCompatible(final EdmPrimitiveType primitiveType) {
-    return primitiveType instanceof Uint7
-        || primitiveType instanceof EdmByte
+    return primitiveType instanceof EdmByte
         || primitiveType instanceof EdmSByte
         || primitiveType instanceof EdmInt16
         || primitiveType instanceof EdmInt32