diff --git a/ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RepositoryResourceProvider.java b/ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RepositoryResourceProvider.java
index edec1eec63..60dff694bb 100644
--- a/ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RepositoryResourceProvider.java
+++ b/ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RepositoryResourceProvider.java
@@ -117,15 +117,25 @@ public class RepositoryResourceProvider extends AbstractControllerResourceProvid
       throws SystemException, UnsupportedPropertyException,
       NoSuchResourceException, NoSuchParentResourceException {
 
-    final Set<RepositoryRequest> requests = new HashSet<>();
+    final Set<RepositoryRequest> requestsToVerifyBaseURLs = new HashSet<>();
 
     Iterator<Map<String,Object>> iterator = request.getProperties().iterator();
     if (iterator.hasNext()) {
       for (Map<String, Object> propertyMap : getPropertyMaps(iterator.next(), predicate)) {
-        requests.add(getRequest(propertyMap));
+        RepositoryRequest rr = getRequest(propertyMap);
+        if(rr.isVerifyBaseUrl()) {
+          requestsToVerifyBaseURLs.add(rr);
+        }
       }
     }
 
+    //Validation only - used by the cluster installation
+    try {
+      getManagementController().verifyRepositories(requestsToVerifyBaseURLs);
+    } catch (AmbariException e) {
+      throw new SystemException("", e);
+    }
+
     return getRequestStatus(null);
   }
 