diff --git a/stack/services/src/main/java/org/usergrid/services/AbstractService.java b/stack/services/src/main/java/org/usergrid/services/AbstractService.java
index fc76ccf99..c9f940022 100644
--- a/stack/services/src/main/java/org/usergrid/services/AbstractService.java
+++ b/stack/services/src/main/java/org/usergrid/services/AbstractService.java
@@ -1,39 +1,17 @@
 /*******************************************************************************
- * Copyright (c) 2010, 2011 Ed Anuff and Usergrid, all rights reserved.
- * http://www.usergrid.com
+ * Copyright 2012 Apigee Corporation
  * 
- * This file is part of Usergrid Stack.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
  * 
- * Usergrid Stack is free software: you can redistribute it and/or modify it
- * under the terms of the GNU Affero General Public License as published by the
- * Free Software Foundation, either version 3 of the License, or (at your option)
- * any later version.
+ *   http://www.apache.org/licenses/LICENSE-2.0
  * 
- * Usergrid Stack is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
- * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
- * details.
- * 
- * You should have received a copy of the GNU Affero General Public License along
- * with Usergrid Stack. If not, see <http://www.gnu.org/licenses/>.
- * 
- * Additional permission under GNU AGPL version 3 section 7
- * 
- * Linking Usergrid Stack statically or dynamically with other modules is making
- * a combined work based on Usergrid Stack. Thus, the terms and conditions of the
- * GNU General Public License cover the whole combination.
- * 
- * In addition, as a special exception, the copyright holders of Usergrid Stack
- * give you permission to combine Usergrid Stack with free software programs or
- * libraries that are released under the GNU LGPL and with independent modules
- * that communicate with Usergrid Stack solely through:
- * 
- *   - Classes implementing the org.usergrid.services.Service interface
- *   - Apache Shiro Realms and Filters
- *   - Servlet Filters and JAX-RS/Jersey Filters
- * 
- * You may copy and distribute such a system following the terms of the GNU AGPL
- * for Usergrid Stack and the licenses of the other code concerned, provided that
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  ******************************************************************************/
 package org.usergrid.services;
 