diff --git a/geode-core/src/main/java/org/apache/geode/internal/logging/Configuration.java b/geode-core/src/main/java/org/apache/geode/internal/logging/Configuration.java
index 566d9dd3ac..f7f2f52ac4 100644
--- a/geode-core/src/main/java/org/apache/geode/internal/logging/Configuration.java
+++ b/geode-core/src/main/java/org/apache/geode/internal/logging/Configuration.java
@@ -15,26 +15,26 @@
 package org.apache.geode.internal.logging;
 
 import static org.apache.geode.internal.lang.SystemPropertyHelper.GEODE_PREFIX;
-import static org.apache.geode.internal.logging.Configuration.LogLevelUpdateOccurs.ONLY_WHEN_USING_DEFAULT_CONFIG;
-import static org.apache.geode.internal.logging.Configuration.LogLevelUpdateScope.GEODE_LOGGERS;
-import static org.apache.geode.internal.logging.InternalLogWriter.CONFIG_LEVEL;
-
-import org.apache.logging.log4j.Logger;
-import org.apache.logging.log4j.status.StatusLogger;
+import static org.apache.geode.internal.logging.LogLevelUpdateOccurs.ONLY_WHEN_USING_DEFAULT_CONFIG;
+import static org.apache.geode.internal.logging.LogLevelUpdateScope.GEODE_LOGGERS;
 
 import org.apache.geode.annotations.TestingOnly;
-import org.apache.geode.internal.ClassPathLoader;
+import org.apache.geode.cache.Cache;
 
 /**
- * Provides logging configuration by managing an {@link ProviderAgent} for the logging backend
- * provider.
+ * Provides logging configuration by managing a {@link ProviderAgent} for the logging backend.
  */
 public class Configuration implements LogConfigListener {
 
-  private static final Logger LOGGER = StatusLogger.getLogger();
-
-  public static final int DEFAULT_LOGWRITER_LEVEL = CONFIG_LEVEL;
+  /**
+   * The default {@link LogWriterLevel} is {@link LogWriterLevel#CONFIG}.
+   */
+  @TestingOnly
+  public static final int DEFAULT_LOGWRITER_LEVEL = LogWriterLevel.CONFIG.intLevel();
 
+  /**
+   * Header for logging the Startup Configuration during {@code Cache} creation.
+   */
   public static final String STARTUP_CONFIGURATION = "Startup Configuration: ";
 
   /**
@@ -42,8 +42,19 @@ public class Configuration implements LogConfigListener {
    */
   public static final String CLI_CONFIG = "/log4j2-cli.xml";
 
+  /**
+   * The root name of all Geode loggers.
+   */
   public static final String GEODE_LOGGER_PREFIX = "org.apache.geode";
+
+  /**
+   * The name of the main Geode logger returned by {@link Cache#getLogger()}.
+   */
   public static final String MAIN_LOGGER_NAME = GEODE_LOGGER_PREFIX;
+
+  /**
+   * The name of the security Geode logger returned by {@link Cache#getSecurityLogger()}.
+   */
   public static final String SECURITY_LOGGER_NAME = GEODE_LOGGER_PREFIX + ".security";
 
   /**
@@ -58,28 +69,6 @@ public class Configuration implements LogConfigListener {
   static final String LOG_LEVEL_UPDATE_SCOPE_PROPERTY =
       GEODE_PREFIX + "LOG_LEVEL_UPDATE_SCOPE";
 
-  /**
-   * System property that may be used to override which {@code ProviderAgent} to use.
-   */
-  static final String PROVIDER_AGENT_NAME_PROPERTY = GEODE_PREFIX + "PROVIDER_AGENT_NAME";
-
-  /**
-   * The default {@code ProviderAgent} is {@code Log4jAgent}.
-   */
-  static final String DEFAULT_PROVIDER_AGENT_NAME =
-      "org.apache.geode.internal.logging.log4j.Log4jAgent";
-
-  /**
-   * The default {@code LogLevelUpdateOccurs} is {@code ONLY_WHEN_USING_DEFAULT_CONFIG}.
-   */
-  private static final LogLevelUpdateOccurs DEFAULT_LOG_LEVEL_UPDATE_OCCURS =
-      ONLY_WHEN_USING_DEFAULT_CONFIG;
-
-  /**
-   * The default {@code LogLevelUpdateScope} is {@code GEODE_LOGGERS}.
-   */
-  private static final LogLevelUpdateScope DEFAULT_LOG_LEVEL_UPDATE_SCOPE = GEODE_LOGGERS;
-
   private final LogLevelUpdateOccurs logLevelUpdateOccurs;
   private final LogLevelUpdateScope logLevelUpdateScope;
 
@@ -91,7 +80,8 @@ public class Configuration implements LogConfigListener {
   private LogConfigSupplier logConfigSupplier;
 
   public static Configuration create() {
-    return create(getLogLevelUpdateOccurs(), getLogLevelUpdateScope(), createProviderAgent());
+    return create(getLogLevelUpdateOccurs(), getLogLevelUpdateScope(), new ProviderAgentLoader()
+        .findProviderAgent());
   }
 
   @TestingOnly
@@ -102,7 +92,8 @@ public class Configuration implements LogConfigListener {
   @TestingOnly
   public static Configuration create(final LogLevelUpdateOccurs logLevelUpdateOccurs,
       final LogLevelUpdateScope logLevelUpdateScope) {
-    return create(logLevelUpdateOccurs, logLevelUpdateScope, createProviderAgent());
+    return create(logLevelUpdateOccurs, logLevelUpdateScope,
+        new ProviderAgentLoader().findProviderAgent());
   }
 
   @TestingOnly
@@ -121,33 +112,19 @@ public class Configuration implements LogConfigListener {
   static LogLevelUpdateOccurs getLogLevelUpdateOccurs() {
     try {
       return LogLevelUpdateOccurs.valueOf(System.getProperty(LOG_LEVEL_UPDATE_OCCURS_PROPERTY,
-          DEFAULT_LOG_LEVEL_UPDATE_OCCURS.name()).toUpperCase());
+          ONLY_WHEN_USING_DEFAULT_CONFIG.name()).toUpperCase());
     } catch (IllegalArgumentException e) {
-      return DEFAULT_LOG_LEVEL_UPDATE_OCCURS;
+      return ONLY_WHEN_USING_DEFAULT_CONFIG;
     }
   }
 
   static LogLevelUpdateScope getLogLevelUpdateScope() {
     try {
       return LogLevelUpdateScope.valueOf(System.getProperty(LOG_LEVEL_UPDATE_SCOPE_PROPERTY,
-          DEFAULT_LOG_LEVEL_UPDATE_SCOPE.name()).toUpperCase());
+          GEODE_LOGGERS.name()).toUpperCase());
     } catch (IllegalArgumentException e) {
-      return DEFAULT_LOG_LEVEL_UPDATE_SCOPE;
-    }
-  }
-
-  static ProviderAgent createProviderAgent() {
-    String agentClassName =
-        System.getProperty(PROVIDER_AGENT_NAME_PROPERTY, DEFAULT_PROVIDER_AGENT_NAME);
-    try {
-      Class<? extends ProviderAgent> agentClass =
-          ClassPathLoader.getLatest().forName(agentClassName).asSubclass(ProviderAgent.class);
-      return agentClass.newInstance();
-    } catch (ClassNotFoundException | ClassCastException | InstantiationException
-        | IllegalAccessException e) {
-      LOGGER.warn("Unable to create ProviderAgent of type {}", agentClassName, e);
+      return GEODE_LOGGERS;
     }
-    return new NullProviderAgent();
   }
 
   /**
@@ -175,17 +152,6 @@ public class Configuration implements LogConfigListener {
     providerAgent.configure(logConfig, logLevelUpdateOccurs, logLevelUpdateScope);
   }
 
-  void enableLoggingToStandardOutput() {
-    providerAgent.enableLoggingToStandardOutput();
-  }
-
-  void disableLoggingToStandardOutputIfLoggingToFile() {
-    LogConfig logConfig = logConfigSupplier.getLogConfig();
-    if (logConfig.getLogFile().exists()) {
-      providerAgent.disableLoggingToStandardOutput();
-    }
-  }
-
   /**
    * Removes this configuration as a {@code LogConfigListener} if it's currently registered and
    * cleans up the {@code ProviderAgent}.
@@ -199,39 +165,26 @@ public class Configuration implements LogConfigListener {
     providerAgent.cleanup();
   }
 
-  @TestingOnly
-  synchronized LogConfigSupplier getLogConfigSupplier() {
-    return logConfigSupplier;
+  String getConfigurationInfo() {
+    return providerAgent.getConfigurationInfo();
   }
 
-  /**
-   * Controls whether or not log level updates should be triggered.
-   */
-  public enum LogLevelUpdateOccurs {
-    NEVER,
-    ONLY_WHEN_USING_DEFAULT_CONFIG,
-    ALWAYS;
-
-    public boolean never() {
-      return this == NEVER;
-    }
-
-    public boolean always() {
-      return this == ALWAYS;
+  void enableLoggingToStandardOutput() {
+    LogConfig logConfig = logConfigSupplier.getLogConfig();
+    if (logConfig.getLogFile().exists()) {
+      providerAgent.enableLoggingToStandardOutput();
     }
+  }
 
-    public boolean onlyWhenUsingDefaultConfig() {
-      return this == ONLY_WHEN_USING_DEFAULT_CONFIG;
+  void disableLoggingToStandardOutputIfLoggingToFile() {
+    LogConfig logConfig = logConfigSupplier.getLogConfig();
+    if (logConfig.getLogFile().exists()) {
+      providerAgent.disableLoggingToStandardOutput();
     }
   }
 
-  /**
-   * Controls the scope of which packages of loggers are updated when the log level changes.
-   */
-  public enum LogLevelUpdateScope {
-    GEODE_LOGGERS,
-    GEODE_AND_SECURITY_LOGGERS,
-    GEODE_AND_APPLICATION_LOGGERS,
-    ALL_LOGGERS
+  @TestingOnly
+  synchronized LogConfigSupplier getLogConfigSupplier() {
+    return logConfigSupplier;
   }
 }