diff --git a/tiles-autotag-core/src/main/java/org/apache/tiles/autotag/model/TemplateSuite.java b/tiles-autotag-core/src/main/java/org/apache/tiles/autotag/model/TemplateSuite.java
index 8ea8f97..5d844a1 100644
--- a/tiles-autotag-core/src/main/java/org/apache/tiles/autotag/model/TemplateSuite.java
+++ b/tiles-autotag-core/src/main/java/org/apache/tiles/autotag/model/TemplateSuite.java
@@ -54,8 +54,9 @@ public class TemplateSuite {
 
     @Override
     public String toString() {
-        return "TemplateSuite\n[documentation=" + documentation + ", name="
-                + name + ", templateClasses=\n" + templateClasses + "]";
+        return "TemplateSuite [name=" + name + ", documentation="
+                + documentation + ", customVariables=" + customVariables
+                + ", templateClasses=" + templateClasses + "]";
     }
 
     public Map<String, String> getCustomVariables() {