diff --git a/src/main/java/org/apache/aurora/scheduler/async/TaskScheduler.java b/src/main/java/org/apache/aurora/scheduler/async/TaskScheduler.java
index 21ea7d2b9..d0fe3e133 100644
--- a/src/main/java/org/apache/aurora/scheduler/async/TaskScheduler.java
+++ b/src/main/java/org/apache/aurora/scheduler/async/TaskScheduler.java
@@ -42,6 +42,7 @@ import com.twitter.common.stats.StatsProvider;
 import com.twitter.common.util.Clock;
 
 import org.apache.aurora.scheduler.HostOffer;
+import org.apache.aurora.scheduler.async.TaskGroups.GroupKey;
 import org.apache.aurora.scheduler.async.preemptor.Preemptor;
 import org.apache.aurora.scheduler.base.Query;
 import org.apache.aurora.scheduler.base.Tasks;
@@ -216,7 +217,8 @@ public interface TaskScheduler extends EventSubscriber {
         AttributeAggregate aggregate = getJobState(store, task.getJob());
         try {
           boolean launched = offerManager.launchFirst(
-              getAssignerFunction(store, new ResourceRequest(task, taskId, aggregate)));
+              getAssignerFunction(store, new ResourceRequest(task, taskId, aggregate)),
+              new GroupKey(task));
 
           if (!launched) {
             // Task could not be scheduled.