diff --git a/components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQConsumer.java b/components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQConsumer.java
index df3f69805f9..501f2a14f01 100644
--- a/components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQConsumer.java
+++ b/components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQConsumer.java
@@ -29,11 +29,12 @@ import org.apache.camel.Processor;
 import org.apache.camel.impl.DefaultConsumer;
 
 public class RabbitMQConsumer extends DefaultConsumer {
-
     ExecutorService executor;
     Connection conn;
     Channel channel;
 
+    private int closeTimeout = 30 * 1000;
+    
     private final RabbitMQEndpoint endpoint;
 
     public RabbitMQConsumer(RabbitMQEndpoint endpoint, Processor processor) {
@@ -43,9 +44,6 @@ public class RabbitMQConsumer extends DefaultConsumer {
 
     @Override
     protected void doStart() throws Exception {
-        super.doStart();
-        log.info("Starting RabbitMQ consumer");
-
         executor = endpoint.createExecutor();
         log.debug("Using executor {}", executor);
 
@@ -76,27 +74,24 @@ public class RabbitMQConsumer extends DefaultConsumer {
 
     @Override
     protected void doStop() throws Exception {
-        super.doStop();
-        log.info("Stopping RabbitMQ consumer");
+        if (channel != null) {
+            log.debug("Closing channel: {}", channel);
+            channel.close();
+            channel = null;
+        }
         if (conn != null) {
-            try {
-                conn.close();
-            } catch (Exception ignored) {
-                // ignored
-            }
+            log.debug("Closing connection: {} with timeout: {} ms.", conn, closeTimeout);
+            conn.close(closeTimeout);
+            conn = null;
         }
-
-        channel = null;
-        conn = null;
-
         if (executor != null) {
-            if (getEndpoint() != null && getEndpoint().getCamelContext() != null) {
-                getEndpoint().getCamelContext().getExecutorServiceManager().shutdownNow(executor);
+            if (endpoint != null && endpoint.getCamelContext() != null) {
+                endpoint.getCamelContext().getExecutorServiceManager().shutdownNow(executor);
             } else {
                 executor.shutdownNow();
             }
+            executor = null;
         }
-        executor = null;
     }
 
     class RabbitConsumer extends com.rabbitmq.client.DefaultConsumer {