diff --git a/geode-core/src/main/java/org/apache/geode/internal/cache/eviction/EvictionListBuilder.java b/geode-core/src/main/java/org/apache/geode/internal/cache/eviction/EvictionListBuilder.java
index 677be499d1..a2f0563902 100644
--- a/geode-core/src/main/java/org/apache/geode/internal/cache/eviction/EvictionListBuilder.java
+++ b/geode-core/src/main/java/org/apache/geode/internal/cache/eviction/EvictionListBuilder.java
@@ -14,89 +14,33 @@
  */
 package org.apache.geode.internal.cache.eviction;
 
-import org.apache.geode.StatisticsFactory;
-import org.apache.geode.cache.EvictionAlgorithm;
-import org.apache.geode.internal.cache.BucketRegion;
-import org.apache.geode.internal.cache.GemFireCacheImpl;
-import org.apache.geode.internal.cache.InternalRegionArguments;
-import org.apache.geode.internal.cache.PartitionedRegion;
-import org.apache.geode.internal.cache.PlaceHolderDiskRegion;
+import java.util.Optional;
+
 import org.apache.geode.internal.lang.SystemPropertyHelper;
 
 public class EvictionListBuilder {
 
-  private final boolean EVICTION_SCAN_ASYNC =
-      SystemPropertyHelper.getProductBooleanProperty(SystemPropertyHelper.EVICTION_SCAN_ASYNC);
-
-  private EvictionAlgorithm algorithm;
-  private Object region;
-  private EvictionController controller;
-  private InternalRegionArguments args;
+  private final boolean evictionScanAsync;
 
-  public EvictionListBuilder(EvictionAlgorithm algorithm) {
-    this.algorithm = algorithm;
-  }
+  private final EvictionController controller;
 
-  /**
-   * @param region PlaceHolderDiskRegion during disk recovery or LocalRegion
-   */
-  public EvictionListBuilder withRegion(Object region) {
-    this.region = region;
-    return this;
-  }
-
-  public EvictionListBuilder withEvictionController(EvictionController evictionController) {
+  public EvictionListBuilder(EvictionController evictionController) {
     this.controller = evictionController;
-    return this;
-  }
-
-  public EvictionListBuilder withArgs(InternalRegionArguments args) {
-    this.args = args;
-    return this;
+    Optional<Boolean> asyncScan =
+        SystemPropertyHelper.getProductBooleanProperty(SystemPropertyHelper.EVICTION_SCAN_ASYNC);
+    evictionScanAsync = asyncScan.orElse(true);
   }
 
   public EvictionList create() {
-    if (algorithm.isLIFO()) {
-      return new LIFOList(getEvictionStats(), getBucketRegion());
+    if (this.controller.getEvictionAlgorithm().isLIFO()) {
+      return new LIFOList(this.controller);
     } else {
-      if (EVICTION_SCAN_ASYNC) {
-        return new LRUListWithAsyncSorting(getEvictionStats(), getBucketRegion());
+      if (evictionScanAsync) {
+        return new LRUListWithAsyncSorting(this.controller);
       } else {
-        return new LRUListWithSyncSorting(getEvictionStats(), getBucketRegion());
+        return new LRUListWithSyncSorting(this.controller);
       }
     }
   }
 
-  private InternalEvictionStatistics getEvictionStats() {
-    InternalEvictionStatistics statistics = null;
-    if (region != null) {
-      if (region instanceof BucketRegion) {
-        if (args != null && args.getPartitionedRegion() != null) {
-          statistics = (InternalEvictionStatistics) args.getPartitionedRegion()
-              .getEvictionController().getStatistics();
-        } else {
-          statistics = new DisabledEvictionStatistics();
-        }
-      } else if (region instanceof PlaceHolderDiskRegion) {
-        statistics = (InternalEvictionStatistics) ((PlaceHolderDiskRegion) region).getPRLRUStats();
-      } else if (region instanceof PartitionedRegion) {
-        statistics = (InternalEvictionStatistics) ((PartitionedRegion) region)
-            .getPRLRUStatsDuringInitialization();
-        if (statistics != null) {
-          PartitionedRegion partitionedRegion = (PartitionedRegion) region;
-          EvictionController evictionController = partitionedRegion.getEvictionController();
-          ((AbstractEvictionController) evictionController).setStatistics(statistics);
-        }
-      }
-    }
-    if (statistics == null) {
-      StatisticsFactory sf = GemFireCacheImpl.getExisting("").getDistributedSystem();
-      statistics = (InternalEvictionStatistics) controller.initStats(region, sf);
-    }
-    return statistics;
-  }
-
-  private BucketRegion getBucketRegion() {
-    return region instanceof BucketRegion ? (BucketRegion) region : null;
-  }
 }