diff --git a/server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java b/server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java
index ff0bf483b..bd1943c07 100644
--- a/server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java
+++ b/server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java
@@ -21,6 +21,7 @@ package org.apache.james.vault;
 
 import java.io.InputStream;
 
+import javax.inject.Inject;
 import javax.mail.MessagingException;
 
 import org.apache.james.core.User;
@@ -42,6 +43,11 @@ import reactor.core.scheduler.Schedulers;
 
 public class MailRepositoryDeletedMessageVault implements DeletedMessageVault {
     public static class Configuration {
+        public static Configuration from(org.apache.commons.configuration.Configuration propertiesConfiguration) {
+            return new Configuration(
+                MailRepositoryUrl.from(propertiesConfiguration.getString("urlPrefix")));
+        }
+
         private final MailRepositoryUrl urlPrefix;
 
         public Configuration(MailRepositoryUrl urlPrefix) {
@@ -53,7 +59,8 @@ public class MailRepositoryDeletedMessageVault implements DeletedMessageVault {
     private final Configuration configuration;
     private final MailConverter mailConverter;
 
-    public MailRepositoryDeletedMessageVault(MailRepositoryStore mailRepositoryStore, Configuration configuration, MailConverter mailConverter) {
+    @Inject
+    MailRepositoryDeletedMessageVault(MailRepositoryStore mailRepositoryStore, Configuration configuration, MailConverter mailConverter) {
         this.mailRepositoryStore = mailRepositoryStore;
         this.configuration = configuration;
         this.mailConverter = mailConverter;