diff --git a/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SerializableCoder.java b/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SerializableCoder.java
index d682345d88..ea84dbdd58 100644
--- a/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SerializableCoder.java
+++ b/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/SerializableCoder.java
@@ -72,6 +72,11 @@ public class SerializableCoder<T extends Serializable> extends CustomCoder<T> {
     return new SerializableCoder<>(clazz, type);
   }
 
+  @Override
+  public boolean consistentWithEquals() {
+    return false;
+  }
+
   /**
    * The structural value of the object is the object itself. The {@link SerializableCoder} should
    * be only used for objects with a proper {@link Object#equals} implementation.