diff --git a/src/java/org/apache/commons/functor/adapter/IgnoreRightProcedure.java b/src/java/org/apache/commons/functor/adapter/IgnoreRightProcedure.java
index 783532b..4972999 100644
--- a/src/java/org/apache/commons/functor/adapter/IgnoreRightProcedure.java
+++ b/src/java/org/apache/commons/functor/adapter/IgnoreRightProcedure.java
@@ -34,7 +34,7 @@ import org.apache.commons.functor.UnaryProcedure;
  * an instance whose delegate is not 
  * <code>Serializable</code> will result in an exception.
  * 
- * @version $Revision: 1.3 $ $Date: 2004/02/28 03:35:23 $
+ * @version $Revision$ $Date$
  * @author Rodney Waldhoff
  */
 public final class IgnoreRightProcedure implements BinaryProcedure, Serializable {