diff --git a/engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java b/engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java
index 27f58f0a9..7e23fa5c8 100644
--- a/engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java
+++ b/engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java
@@ -659,7 +659,6 @@ public class StramClientUtils
       engine.eval("var _prop = {}");
       for (Map.Entry<String, String> entry : vars) {
         String evalString = String.format("_prop[\"%s\"] = \"%s\"", StringEscapeUtils.escapeJava(entry.getKey()), StringEscapeUtils.escapeJava(entry.getValue()));
-        LOG.debug("Evaluating: {}", evalString);
         engine.eval(evalString);
       }
     } catch (ScriptException ex) {
@@ -709,6 +708,18 @@ public class StramClientUtils
     }
   }
 
+  public static void evalConfiguration(Configuration conf)
+  {
+    Properties props = new Properties();
+    for (Map.Entry entry : conf) {
+      props.put(entry.getKey(), entry.getValue());
+    }
+    evalProperties(props, conf);
+    for (Map.Entry<Object, Object> entry : props.entrySet()) {
+      conf.set((String)entry.getKey(), (String)entry.getValue());
+    }
+  }
+
   public static <T> T doAs(String userName, PrivilegedExceptionAction<T> action) throws Exception
   {
     if (StringUtils.isNotBlank(userName) && !userName.equals(UserGroupInformation.getLoginUser().getShortUserName())) {